src/main/kotlin/org/rust/lang/core/completion/LookupElements.kt

```diff
@@ -20,18 +20,33 @@ import org.rust.lang.core.types.ty.TyUnknown
 import org.rust.lang.core.types.type
 
 const val KEYWORD_PRIORITY = 10.0
+private const val ENUM_VARIANT_PRIORITY = 4.0
+private const val FIELD_DECL_PRIORITY = 3.0
+private const val INHERENT_IMPL_MEMBER_PRIORITY = 2.0
+private const val DEFAULT_PRIORITY = 0.0
 private const val MACRO_PRIORITY = -0.1
 
 fun createLookupElement(element: RsElement, scopeName: String): LookupElement {
     val base = element.getLookupElementBuilder(scopeName)
         .withInsertHandler { context: InsertionContext, _ -> getInsertHandler(element, scopeName, context) }
 
-    if (element is RsMacro) return base.withPriority(MACRO_PRIORITY)
+    val priority = when (element) {
+        is RsMacro -> MACRO_PRIORITY
+        is RsEnumVariant -> ENUM_VARIANT_PRIORITY
+        is RsFieldDecl -> FIELD_DECL_PRIORITY
+        is RsAbstractable -> {
+            val owner = element.owner
+            when {
+                owner is RsAbstractableOwner.Impl && owner.isInherent -> INHERENT_IMPL_MEMBER_PRIORITY
+                else -> DEFAULT_PRIORITY
+            }
+        }
+        else -> DEFAULT_PRIORITY
+    }
 
-    return base
+    return if (priority == DEFAULT_PRIORITY) base else base.withPriority(priority)
```

Comment:
Is this check really necessary? If so, it could be moved into `LookupElementBuilder.withPriority`. Otherwise, use just `return base.withPriority(priority)`.

Reply:
This check allows to get rid of unnecessary allocation if we don't change priority. I moved it to `withPriority`